### ISSUE TITLE
Sometimes xbmc fails to change the resolution (refresh rate)

### ISSUE THREAD

**Issue by andrasbiro** [xbmc-imx6/xbmc#63]
It's easy to reproduce on my system: xbmc gui runs on 60Hz. I repedatly play 24p videos and stop them. It will fail eventually just after a few success (not more than 10). If it fails, I only see the arch login prompt, but I hear that xbmc works in the background. I can even start and stop videos blindly, sometimes it recovers the screen, but usually not.

Failure looks like this in the xbmc.log:
 DEBUG: SetNativeResolution: S:1920x1080p-60
14:15:34 T:1958981632  NOTICE: CreateWindow: Could not create a surface. Trying with a fresh Native Window.
14:15:34 T:1958981632   ERROR: EGL error in CreateSurface: 300b
14:15:34 T:1958981632   ERROR: CreateWindow: Could not create surface
14:15:34 T:1958981632   ERROR: CreateNewWindow: Could not create new window

While it should have look like this:
14:18:08 T:1958649856   DEBUG: SetNativeResolution: S:1920x1080p-60
14:18:08 T:1958649856  NOTICE: CreateWindow: Could not create a surface. Trying with a fresh Native Window.
14:18:08 T:1821463584   DEBUG: CActiveAE - device change event
14:18:08 T:1958649856   DEBUG: CreateWindow: Created surface of size 1920x1080

My system: cubox-i4, Panasonic TX-L47ET60E, arch linux arm, with kernel from Stephan' linaro repo (but it had this problem on the solidrun kernel as well).

**Comment by koying** [comment]
Yeah, I confirm this (although it occurs very rarely to me those days).
I'm pretty sure it's not related to XBMC, though.

What also occurs to me sometimes is that I cannot get EGL to kick in even when starting XBMC. Unplugging / plugging the cuboxi does not help.
Only if I leave it unplugged for some minutes do I succeed to start it again.

I suspect an hardware heat issue on the GPU...

**Comment by andrasbiro** [comment]
@koying I disagree. I don't have the EGL start problem (at least I don't have it when the screen is on. it does funny things sometimes if the screen is off, but I didn't bother with it). Maybe this is indeed hw related.
But the bug in my case seems very likely xbmc or kernel bug to me. Sometimes it happens on the first video, after the tv was off for a couple of hours, and the cubox just showed the dimmed down confluence menu. I think the play-stop loop shouldn't heat the gpu that much anyway. And something should fail if I change the xbmc settings to keep the 60Hz output if it's hw bug, but it works flowlessly.

**Comment by andrasbiro** [comment]
I think I found the solution, this is in fact an arch linux bug (well, I'm not sure if it's even a bug or a user error with configuration).

So, arch linux doesn't set fbmem and gpumem in uEnv.txt, I think that caused the problem, at least I couldn't reproduce it that easily But this raises the question: What is the ideal setting for xbmc?  I set it to 18M/64M, mostly based on [this](https://community.freescale.com/docs/DOC-93591), but this is for android, where triple buffering is (i think) default. (Screen is 1920x1080@60, RGB24)

If I'm right about the solution, [geexbox](http://git.openbricks.org/?p=openbricks.git;a=blob;f=config/platforms/arm/imx6/machines/cuboxi/uEnv.txt;h=abfa19a6df91a8e743b9377ad99064c6aaf788a4;hb=HEAD) and [openelec](https://github.com/OpenELEC/OpenELEC.tv/blob/master/projects/Cuboxi/bootloader/uEnv.txt) has the exact same problem.

**Comment by andrasbiro** [comment]
18M/64M was still unstable, so I set it to 32M/256M (which is probably way too much, but I had more than enough RAM anyway), with that setting, it's stable.

**Comment by CrawX** [comment]
Running the newest version of the kernel and the libraries (3.10.17ga) on the wandboard, I can not reproduce this bug specifically (I tried starting the video ~15times, worked fine everytime).

However, I sometimes have the same behaviour you described (dimmed down screen, xbmc not visible but running...).

Could you post the exact setting you applied to uEnv.txt? It seems like the values can be set independently for each framebuffer. I'd be happy to give it a shot, altough I can not really reproduce this bug. I guess I'll have to wait for it to happen :)

Cheers

**Comment by andrasbiro** [comment]
Actually, it just did this again a few hours age. But this is really rare now, I played a lot of videos in the last week (I tested boblight), and it worked flawlessly.

So, uEnv.txt:
mmcroot=/dev/mmcblk0p2 rootwait rw console=ttymxc0,115200 console=tty1 consoleblank=0 video=mxcfb0:dev=hdmi,1920x1080M@60,if=RGB24,bpp=16 dmfc=3 fbmem=32M gpumem=256M
[andris@cuboxi4 ~]$ mmcroot=/dev/mmcblk0p2 rootwait rw console=ttymxc0,115200 console=tty1 consoleblank=0 video=mxcfb0:dev=hdmi,1920x1080M@60,if=RGB24,bpp=16 dmfc=3 fbmem=32M gpumem=256M
kernel source pkg:
https://dl.dropboxusercontent.com/u/363226/linux-imx6-cubox-dt-3.10.30-16.src.tar.gz
(almost the official, but with Wolfgar's tree, one patch was removed, an other one added, and the config was slightly modified to support the mcp2200 usb-uart bridge).
everything else is "official", but xbmc has this patch: https://github.com/archlinuxarm/PKGBUILDs/pull/821

**Comment by andrasbiro** [comment]
Whoops, sorry, I didn't realize the library/firmware update, I didn't updated it until now.

**Comment by smallint** [comment]
Does the issue persist on your side or is it resolved with the above uEnv.txt? I installed Arch on my cubox and I am facing the same issue again that I had on my Wandboard with kernel 3.0.17. Sometimes the resolution switch ends up in the console and xbmc keeps running on another framebuffer. Only restarting xbmc helps. I could minimize that effect by settings a pause of 2s after the resolution switch in xbmc itself. But it is still not perfect. Even sometimes after booting the xbmc splash does not go away but xbmc is running (I hear the sounds).

**Comment by andrasbiro** [comment]
I still have this issue, but very-very rearly. I also had the splash screen issue, but I didn't realize it was the same.
However, I'm using the 3.10.30 kernel.

**Comment by wolfgar** [comment]
Hi
I also confirm this issue (whatever kernel and/or boards)
I guess it is so rare (empirically I would tell 1 time out of 100) that nobody really cares that much about it ;)
Stephan

**Comment by Apokalypzx** [comment]
```
I've been having this problem since I got my cubox-i. I'm on the new 3.14 kernel in arch and still have the problem.
I think most people don't care about it because they probably don't select the "Match refresh rate to video" xbmc option selected. Either way, it's really more of a nuisance than anything. I just setup a hotkey on my remote to restart xbmc, so when it happens I just press a button.
Anyway, thought I would comment to make it known, this is still a problem.
```
